File: demo-spring-boot/demo-spring-boot-mvc/src/main/java/com/future/demo/controller/Person.java
package com.future.demo.controller;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
public class Person {
    Long id;
    private String name;
    private int age;
    private String [] hobby;
    private Address address;

    @Data
    @AllArgsConstructor
    @NoArgsConstructor
    public static class Address {
        private String city;
        private String country;
    }
}

File: demo-c++/demo-c-plus-class/person.cpp
#include "person.h"
#include <iostream>

Person::Person(std::string name, int age) {
    this->name = name;
    this->age = age;
}

Person::~Person() {

}

void Person::sayHi() {
    std::cout << "Person say hi to " << this->name << "(age=" << this->age << ")" << std::endl;
}
